test/proc/race.js

```diff
@@ -110,3 +110,22 @@ test('processor race between sync effects', assert => {
     assert.end();
   })
 });
+
+test('processor race error has origin', assert => {
```

Comment:
most tests do not put assert calls inside a generator, but rather have `actual` and `expected` arrays which are compared at the end, so basically that would look something like
```
test('processor race error has origin', assert => {
  assert.plan(1);
  const actual = []
  const expected = ['willErr']

  function* genFn() {
    try {
      yield io.race({
        wontResolve: new Promise((resolve, reject) => {}),
        willErr: Promise.reject(new Error("FAIL")),
      })
    } catch(err) {
      actual.push(err[RACE_ORIGIN])
    }
  }

  proc(genFn()).done.then(() => {
    assert.deepEqual(actual, expected, 'thrown error should have RACE_ORIGIN set on itself')
  })
})
```

sometimes smth like `Promise.resolve(1)` is used in test to cause asynchrony, no need for that here though

this should also be accompanied by a test which do not set `RACE_ORIGIN`, when something else than an Error is thrown